### ISSUE TITLE
Windows affected by Stage Manager on macOS Sonoma continue rendering borders

### ISSUE THREAD

**Issue by bobbyl140** [FelixKratz/JankyBorders#29]
I am on macOS Sonoma and using Stage Manager on macOS Sonoma, and I just installed JankyBorders earlier today. I have noticed that when a window is "minimized" (in the column on the left), the border is still in the space where the active window is. I will comment a screenshot once I'm back on my MacBook.

**Comment by FelixKratz** [comment]
I have not used StageManager yet, but I think this is due to the window being transformed by StageManager, rather than properly marked as hidden.

This will require some digging on how to detect when a window is in StageManager reliably, in order to properly hide the border.

**Comment by bobbyl140** [comment]
That makes sense. In the early days of Stage Manager, screen recording and screen sharing apps showed these windows as pixelated and "bent" if you will, like the user sees, instead of showing them like normal. This has been fixed in many apps by now though, which says it is somehow possible.

**Comment by FelixKratz** [comment]
Until this is fixed you can set the inactive color to be fully transparent, such that no window in StageManager renders a border:

```bash
borders inactive_color=0x00000000
```

**Comment by bobbyl140** [comment]
+1 I didn't think of that, that you for pointing it out.

**Comment by FelixKratz** [comment]
This issue is an issue with how transformed windows are handled and reported to us by macOS. If a window is transformed by the window server, we are not notified and I think there is no way of being notified about that, such that we can not react to a window transformation properly. The only solution to this would be to poll the windows transform. But I don't want to do that for performance reasons.

Any further ideas are welcome.

**Comment by ethanuppal** [comment]
> ```shell
> borders inactive_color=0x00000000
> ```

This is great! It might be good idea to add this to the README.

**Comment by Bellavene** [comment]
> This issue is an issue with how transformed windows are handled and reported to us by macOS. If a window is transformed by the window server, we are not notified and I think there is no way of being notified about that, such that we can not react to a window transformation properly. The only solution to this would be to poll the windows transform. But I don't want to do that for performance reasons.
> 
> Any further ideas are welcome.

I don't think that this is a problem at all, stage manager in current implementation is not great anyway (no shortcuts, not scriptable, wasted screen space). And most people, who use it, don't need any borders anyway.
